catalyst/contrib/dl/callbacks/cmc_callback.py

```diff
@@ -0,0 +1,113 @@
+from typing import List, Optional
+
+import torch
+
+from catalyst.dl import Callback
+
+
+def _cmc_score_count(
+    distances: torch.Tensor, conformity_matrix: torch.Tensor, topk: int = 1,
+) -> float:
+    """
+    More convenient to write tests with distance_matrix
+    Args:
+        distances: distance matrix shape of (n_embeddings_x, n_embeddings_y)
+        conformity_matrix: binary matrix with 1 on same label pos and 0 otherwise
+        topk: number of top examples for cumulative score counting
+
+    Returns:
+        cmc score
+    """
+    position_matrix = torch.argsort(distances, dim=1)
+    conformity_matrix = conformity_matrix.type(torch.bool)
+    position_matrix[~conformity_matrix] = (
+        topk + 1
+    )  # value large enough not to be counted
+
+    closest = position_matrix.min(dim=1)[0]
+    k_mask = (closest < topk).type(torch.float)
+    return k_mask.mean().item()
+
+
+def cmc_score(
+    query_embeddings: torch.Tensor,
+    gallery_embeddings: torch.Tensor,
+    conformity_matrix: torch.Tensor,
+    topk: int = 1,
+) -> float:
+    """
+
+    Args:
+        query_embeddings: tensor shape of (n_embeddings, embedding_dim)
+            embeddings of the objects in querry
+        gallery_embeddings: tensor shape of (n_embeddings, embedding_dim)
+            embeddings of the objects in gallery
+        conformity_matrix: binary matrix with 1 on same label pos
+            and 0 otherwise
+        topk: number of top examples for cumulative score counting
+
+    Returns:
+        cmc score
+    """
+    distances = torch.cdist(query_embeddings, gallery_embeddings)
+    return _cmc_score_count(distances, conformity_matrix, topk)
+
+
+class CMCScoreCallback(Callback):
+    """Cumulative Matching Characteristics"""
```

Comment:
and docs here?

Reply:
maybe, it's good idea to add minimal example on how to use this callback with MNIST-lite dataset?

Reply:
Docs like [here](https://catalyst-team.github.io/catalyst/api/dl.html#module-catalyst.dl.callbacks.metrics.accuracy)?)